src/uu/ls/src/ls.rs

```diff
@@ -2689,8 +2715,19 @@ fn display_item_long(
 
         write!(output_display, " {} ", display_date(md, config)).unwrap();
 
-        let displayed_item =
+        let item_name =
             display_item_name(item, config, None, String::new(), out, style_manager).contents;
+
+        let displayed_item = if quoted {
+            if item_name.starts_with('\'') {
+                item_name
+            } else {
+                format!(" {}", item_name)
+            }
+        } else {
+            item_name
+        };
```

Comment:
I think you can simplify this snippet by using a single `if`.

The rest of the changes look good.

Reply:
Yep!